packages/components/src/site-thumbnail/index.tsx

```diff
@@ -1,54 +1,75 @@
+import { css, CSSObject } from '@emotion/css';
 import classnames from 'classnames';
 import { ReactNode } from 'react';
 import './style.scss';
 import { MShotsOptions, useMshotsImg } from './use-mshots-img';
 import { getTextColorFromBackground } from './utils';
 
-const MSHOTS_OPTION: MShotsOptions = {
-	vpw: 1200,
-	vph: 1200,
-	w: 374 * 2,
-};
+export type SizeCss = { width: number; height: number };
+
+const DEFAULT_SIZE = { width: 106, height: 76.55 };
+
+const VIEWPORT_BASE = 1200;
 
 type Props = {
+	alt: string;
 	backgroundColor?: string;
-	className?: string;
+	style?: CSSObject;
 	mShotsUrl?: string;
-	size?: 'small' | 'medium';
+	width?: number;
+	height?: number;
+	dimensionsSrcset?: Array< SizeCss >;
+	sizesAttr?: string;
 	children?: ReactNode;
-	alt?: string;
 	bgColorImgUrl?: string;
+	viewport?: number;
 	mshotsOption?: MShotsOptions;
 };
 
 export const SiteThumbnail = ( {
 	backgroundColor,
 	children,
-	className,
+	style,
 	alt,
 	mShotsUrl = '',
 	bgColorImgUrl,
-	size = 'small',
-	mshotsOption = MSHOTS_OPTION,
+	width = DEFAULT_SIZE.width,
+	height = DEFAULT_SIZE.height,
+	dimensionsSrcset = [],
+	sizesAttr = '',
+	viewport = VIEWPORT_BASE,
+	mshotsOption,
 }: Props ) => {
-	const { src, isLoading, isError, imgRef } = useMshotsImg( mShotsUrl, mshotsOption );
+	const options: MShotsOptions = {
+		vpw: viewport,
+		vph: viewport,
+		w: width,
+		h: height,
+		...mshotsOption,
+	};
+	const { imgProps, isLoading, isError, imgRef } = useMshotsImg( mShotsUrl, options, [
+		...dimensionsSrcset,
+		{ width, height },
+	] );
 
 	const color = backgroundColor && getTextColorFromBackground( backgroundColor );
 
 	const classes = classnames(
 		'site-thumbnail',
-		className,
 		isLoading ? 'site-thumbnail-loading' : 'site-thumbnail-visible',
-		`site-thumbnail__size-${ size }`
+		// default image width, height given by dimension if not specified in style
+		css( { width, height }, style )
```

Comment:
I don't think I like this pattern of generating CSS class names at render time. All of the thumbnails happen to have matching widths and heights, and so luckily enough we're only generating one CSS class that they all use. But theoretically, it's possible that every `<SiteThumbnail>` could end up generating its own class.

I think because the `<SiteThumbnail>`'s prop was called `style` I was expecting that it'd be used to set an inline style on the element.

Emotion does have support for this sort of thing though—where you pass an object of styles down to a child component—using [the `css` prop](https://emotion.sh/docs/css-prop). But we don't use it because no one has set up the build tooling that it requires yet.

Reply:
The issue I tried to fix using the CSSObject instead of the className was that, in some cases the classname  prop had some collisions with the className defined inside the component `scss`  `site-thumbnail__size-${ size }`.
Receiving the CSSObject, and generating the className on the fly I was making sure that the style from props had preference over other styles.

In general I like the freedom to re-style a component just passing the styles form the parent, and you don't get that kind of freedom using static strings `classNames`.

I also agree that in case we don't need to have this freedom, because every thumbnail should look the same.  So I've refactored the code to use the `className` instead of the CSSObject 👍 . Thanks for the suggestion 🙌 . ([b6832e995d5d815d81297bb056f244b92c930969](https://github.com/Automattic/wp-calypso/pull/66988/commits/b6832e995d5d815d81297bb056f244b92c930969))